pandas/core/arrays/sparse/accessor.py

```diff
@@ -228,14 +228,29 @@ def from_spmatrix(cls, data, index=None, columns=None):
         2  0.0  0.0  1.0
         """
         from pandas import DataFrame
+        from pandas._libs.sparse import IntIndex
 
         data = data.tocsc()
         index, columns = cls._prep_index(data, index, columns)
```

Comment:
I think the problem is that this `_pre_index` doesn't actually return Index objects when columns or index is not None

Reply:
OK, I see. Thanks for confirming that passing duplicate columns names in an Index object is expected to work in `_from_arrays`. Will look into it. Thanks Joris!

Reply:
That seems to be it, as doing `pd.DataFrame.sparse.from_spmatrix(mat, columns=pd.Index(['a', 'a']))` instead of `pd.DataFrame.sparse.from_spmatrix(mat, columns=['a', 'a'])` works. 

You can add in here a `ensure_index` call. (`from pandas.core.indexes.api import ensure_index`)